### PR TITLE
Bellatrix API support for block endpoints

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Adds bellatrix support for the following API endpoints:
- https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockV2
- https://ethereum.github.io/beacon-APIs/#/Beacon/publishBlock
- https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockAttestations

Non-SSZ functionality for getting blocks had already been supported, but I refactored it.

### CODE REVIEW

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 1]
```diff
@@ -221,7 +219,44 @@ func (bs *Server) SubmitBlock(ctx context.Context, req *ethpbv2.SignedBeaconBloc
...
+		if err := bs.Broadcaster.Broadcast(ctx, wrappedBellatrixBlk.Proto()); err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not broadcast block: %v", err)
+		}
+
+		if err := bs.BlockReceiver.ReceiveBlock(ctx, wrappedBellatrixBlk, root); err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not process beacon block: %v", err)
+		}
+	}
+
+	return &emptypb.Empty{}, status.Error(codes.Internal, "Unknown block type")
```
Comment (terencechain) [inline beacon-chain/rpc/eth/beacon/blocks.go:259]: The amount of duplication on a per upgrade basis is not going to scale nicely. We probably can dedup these similar to this PR: https://github.com/prysmaticlabs/prysm/pull/10297

I understand this is out of scope so feel free to do it later

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 2]
```diff
@@ -274,12 +309,17 @@ func (bs *Server) GetBlockV2(ctx context.Context, req *ethpbv2.BlockRequestV2) (
...
 	defer span.End()
 
-	blk, phase0Blk, err := bs.blocksFromId(ctx, req.BlockId)
+	blk, err := bs.blockFromBlockID(ctx, req.BlockId)
 	err = handleGetBlockError(blk, err)
 	if err != nil {
 		return nil, err
 	}
-	if phase0Blk != nil {
+
```
Comment (terencechain) [inline beacon-chain/rpc/eth/beacon/blocks.go:317]: This seems a little bit anti-pattern by skipping certain errors and returning if the error is nil. What do you think about using a switch statement for it? 

```go
	switch b := blk.(type) {
	case *eth.SignedBeaconBlock:
		
	case *eth.SignedBeaconBlockAltair:
		
	case *eth.SignedBeaconBlockBellatrix:
		
	default:
		return nil, errors.Wrapf(ErrUnsupported, "unable to get block of type %T", i)
	}
```

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 3]
```diff
@@ -352,23 +404,54 @@ func (bs *Server) GetBlockSSZV2(ctx context.Context, req *ethpbv2.BlockRequestV2
...
-	}
-	v2Blk, err := migration.V1Alpha1BeaconBlockAltairToV2(altairBlk.Block)
-	if err != nil {
+	if err != nil && !errors.Is(err, wrapper.ErrUnsupportedAltairBlock) {
 		return nil, status.Errorf(codes.Internal, "Could not get signed beacon block: %v", err)
 	}
-	data := &ethpbv2.SignedBeaconBlockAltair{
-		Message:   v2Blk,
-		Signature: blk.Signature(),
+	if err == nil {
```
Comment (terencechain) [inline beacon-chain/rpc/eth/beacon/blocks.go:412]: Same. Curious whether using switch statement has been considered

**Review by terencechain — APPROVED** [review]

File: beacon-chain/rpc/eth/beacon/blocks.go (hunk 2 above)
Comment (rkapka) [reply beacon-chain/rpc/eth/beacon/blocks.go:317]: I would have to switch on the result of `blk, err := bs.blockFromBlockID(ctx, req.BlockId)`. Unfortunately only one implementation of the interface, `Phase0SignedBeaconBlock`, is exported. The other two are not. Additionally I feel like the interface methods were introduced for a reason and using them here makes sense to me. I agree that the flow is a bit unorthodox though. I reordered some statements and added comments.

**Review by terencechain — APPROVED** [review]

**Pull request merged**